frontend/pages/teams/question/[slug].tsx

```diff
@@ -0,0 +1,276 @@
+import CommentForm from '@/components/organisms/CommentForm'
+import SortDropdown from '@/components/molecules/SortDropdown'
+import AnswerForm from '@/components/organisms/AnswerForm'
+import AnswerDetail from '@/components/organisms/AnswerDetail'
+import Comment from '@/components/organisms/Comment'
+import QuestionDetail from '@/components/organisms/QuestionDetail'
+import GET_QUESTION from '@/helpers/graphql/queries/get_question'
+import { loadingScreenShow } from '@/helpers/loaderSpinnerHelper'
+import { errorNotify } from '@/helpers/toast'
+import { useQuery } from '@apollo/client'
+import { useRouter } from 'next/router'
+import { Fragment, useEffect, useState } from 'react'
+import { FilterType } from '../../index'
+
+export type UserType = {
+    id?: number
+    slug?: string
+    first_name?: string
+    last_name?: string
+    avatar?: string
+}
+
+export type CommentType = {
+    id: number
+    content: string
+    user: UserType
+    created_at: string
+    updated_at: string
+}
+
+export type TagType = {
+    id: number
+    slug: string
+    name: string
+    description: string
+    is_watched_by_user: boolean
+    count_tagged_questions: number
+    count_watching_users: number
+}
+
+export type AnswerType = {
+    id: number
+    content: string
+    created_at: string
+    vote_count: number
+    humanized_created_at: string
+    is_bookmarked: boolean
+    is_correct: boolean
+    is_created_by_user: boolean
+    user_vote: number
+    user: UserType
+    is_from_user: boolean
+    comments: CommentType[]
+    question: { slug: string }
+}
+
+export type QuestionType = {
+    id: number
+    title: string
+    slug: string
+    content: string
+    created_at: string
+    vote_count: number
+    views_count: number
+    humanized_created_at: string
+    tags: TagType[]
+    is_bookmarked: boolean
+    is_from_user: boolean
+    is_answered: boolean
+    user_vote: number
+    user: UserType
+    answers: AnswerType[]
+    comments: CommentType[]
+}
+
+export type AnswerEditType = {
+    id: number | null
+    content: string | null
+}
+
+type RefetchType = {
+    slug?: string
+    shouldAddViewCount: boolean
+    answerSort?: { column: string; order: string }[]
+}
```

Comment:
I think no need to redefined these Types, you can just import them since they are exported from the other file